gpflow/likelihoods/likelihoods.py

```diff
@@ -68,93 +70,323 @@ def inv_probit(x):
     return 0.5 * (1.0 + tf.math.erf(x / np.sqrt(2.0))) * (1 - 2 * jitter) + jitter
 
 
-class Likelihood(Module):
-    def __init__(self):
+class Likelihood(Module, metaclass=abc.ABCMeta):
+    def __init__(self, latent_dim: int, observation_dim: int):
+        """
+        A base class for likelihoods, which specifies an observation model 
+        connecting the latent functions ('F') to the data ('Y').
+
+        All of the members of this class are expected to obey some shape conventions, as specified
+        by latent_dim and observation_dim.
+
+        If we're operating on an array of function values 'F', then the last dimension represents
+        multiple functions (preceding dimensions could represent different data points, or
+        different random samples, for example). Similarly, the last dimension of Y represents a
+        single data point. We check that the dimensions are as this object expects.
+
+        The return shapes of all functions in this class is the broadcasted shape of the arguments,
+        excluding the last dimension of each argument.
+
+        :param latent_dim: the dimension of the vector F of latent functions for a single data point
+        :param observation_dim: the dimension of the observation vector Y for a single data point
+        """
         super().__init__()
         self.num_gauss_hermite_points = 20
```

Comment:
This should probably move into the ScalarLikelihood subclass ?

Reply:
I don't think so.
General shape info and input should be in the base

Reply:
I meant the num_gauss_hermite_points. This base class does not know about quadrature...